src/deepsparse/pipeline.py

```diff
@@ -118,35 +123,39 @@ class PipelineImplementation(Pipeline):
         other runtime information that will be used across instances of the
         MultiModelEngine to provide optimal performance when running multiple
         models concurrently
-    :param threadpool: An optional ThreadPoolExecutor() object, if given the
+    :param executor: A optional ThreadPoolExecutor() object, if given the
         pipeline executes inference requests in a non-blocking manner and returns
-        a Future object, call Future.result() on returned object to get the result
+        a Future object, call Future.result() on returned object to get the result.
+        Can also accept an int number of workers, a ThreadPoolExecutor object is
+        auto-initialized with the specified integer in that case; if True a
+        ThreadPoolExecutor is initialized with min(32, (os.cpu_count() or 1) + 4)
+        max workers. False and None represent synchronous execution
     """
 
     def __init__(
         self,
         model_path: str,
         engine_type: str = DEEPSPARSE_ENGINE,
-        batch_size: int = 1,
+        batch_size: Optional[int] = 1,
         num_cores: int = None,
         scheduler: Scheduler = None,
         input_shapes: List[List[int]] = None,
         alias: Optional[str] = None,
         context: Optional[Context] = None,
-        threadpool: Optional[ThreadPoolExecutor] = None,
+        executor: Optional[Union[ThreadPoolExecutor, int, bool]] = None,
```

Comment:
Changed the name to `executor` from `threadpool`, the idea being that the executor can be a `bool`, `int`, or a `ThreadPoolExecutor` object; Another name that was a consideration was `async` but that's a protected keyword in python, an alternative could be `async_` but I would like to avoid trailing underscores in `__init__`, any other naming suggestions are welcome

Reply:
let's get rid of bool, don't see a benefit to bool over int